### PR TITLE
E2E tests: Move Search tests into Search plugin folder

### PR DESCRIPTION
**Pull request by adimoldovan**
#### Changes proposed in this Pull Request:

Moved search end-to-end tests into a standalone test package.

#### Jetpack product discussion
pd5faL-3G-p2

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Check CI:
	- All tests should pass
	- Search tests should run in a separate job
	- Jetpack job should not contain any search tests
	- See https://github.com/Automattic/jetpack/actions/runs/2004371534

### CODE REVIEW

File: projects/plugins/search/tests/e2e/README.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+```
+
+### Test configuration
+
+The tests rely on an encrypted configuration file, which is included in the [e2e commons package](../../../../../tools/e2e-commons) config folder as [`encrypted.enc`](../../../../../tools/e2e-commons/config/encrypted.enc). To be able to run tests - that file should be decrypted first.
+
+To decrypt the config file (a8c only):
+
+- Find a decryption key. Search secret store for "E2E Jetpack CONFIG_KEY"
+- From the Jetpack Boost E2E tests directory, run `CONFIG_KEY=YOUR_KEY pnpm config:decrypt`. This command should create a new file `local.cjs` in the Jetpack Boost E2E tests config folder.
```
Comment (jsnmoon) [inline projects/plugins/search/tests/e2e/README.md:46]:
```suggestion
- From the Jetpack Search E2E tests directory, run `CONFIG_KEY=YOUR_KEY pnpm config:decrypt`. This command should create a new file `local.cjs` in the Jetpack Search E2E tests config folder.
```

**Review by jsnmoon — DISMISSED** [review]
Works well, thank you!

**Review by kangzj — DISMISSED** [review]
Hi @adimoldovan  Thanks for the PR. Great job it works for me 👍 

I have a question tho - we'll probably need the test set to be run under the following circumstances:
- Jetpack plugin enabled only
- Search plugin enabled only
- Jetpack + Search plugin both enabled

How can we accomplish that? And also could you help us with it? 😄 

Thanks

File: projects/plugins/search/tests/e2e/README.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
+[![Reports status](https://img.shields.io/website?down_color=grey&down_message=Dashboard%20offline&style=for-the-badge&label=E2E%20TEST%20REPORTS&up_color=green&up_message=see%20dashboard&url=https%3A%2F%2Fautomattic.github.io%2Fjetpack-e2e-reports%2F%23%2F)](https://automattic.github.io/jetpack-e2e-reports)
```
Comment (brbrr) [inline projects/plugins/search/tests/e2e/README.md:1]: Just wondering, how useful it is to have similar README files for every plugin that has it's own e2e suite. There are definitely benefits (such as how to run this particular suite), but I'm not sure about the technical bits: docker, tunneling, config, etc. Mainly, I am concerned about the need to maintain these similar docs.

**Review by brbrr — DISMISSED** [review]
Looks good overall! 

Lets apply Jason's suggestion before merging tho. Also, I'm not sure about the need to full featured README here, but this is not blocking/critical.

**Review by kraftbj — DISMISSED** [review]
Agreed with Yarik about not needing to repeat everything in every e2e readme and agreed we can discuss/PR separately.

**Comment by adimoldovan** [comment]
> 
> I have a question tho - we'll probably need the test set to be run under the following circumstances:
> 
> * Jetpack plugin enabled only
> * Search plugin enabled only
> * Jetpack + Search plugin both enabled
> 
> How can we accomplish that? And also could you help us with it? 😄
> 
> Thanks

Hi @kangzj, I will be happy to help with this. 
The tests have access to the wp-cli, so the setup should be pretty straightforward.

File: projects/plugins/search/tests/e2e/README.md (hunk 2 above)
Comment (adimoldovan) [reply projects/plugins/search/tests/e2e/README.md:1]: Yeah, I agree we can optimize the doc files further. I will address this later on.

**Review by brbrr — APPROVED** [review]

**Comment by adimoldovan** [comment]
r242225-wpcom

**Pull request merged**